service/src/main/kotlin/no/nav/su/se/bakover/service/dokument/DistribuerDokumentService.kt

```diff
@@ -0,0 +1,85 @@
+package no.nav.su.se.bakover.service.dokument
+
+import arrow.core.Either
+import arrow.core.right
+import no.nav.su.se.bakover.client.dokdistfordeling.DokDistFordeling
+import no.nav.su.se.bakover.common.journal.JournalpostId
+import no.nav.su.se.bakover.domain.brev.BrevbestillingId
+import no.nav.su.se.bakover.domain.brev.Distribusjonstidspunkt
+import no.nav.su.se.bakover.domain.brev.Distribusjonstype
+import no.nav.su.se.bakover.domain.brev.KunneIkkeBestilleBrevForDokument
+import no.nav.su.se.bakover.domain.brev.KunneIkkeDistribuereBrev
+import no.nav.su.se.bakover.domain.dokument.DokumentRepo
+import no.nav.su.se.bakover.domain.dokument.Dokumentdistribusjon
+import no.nav.su.se.bakover.domain.eksterneiverksettingssteg.KunneIkkeJournalføreOgDistribuereBrev
+import no.nav.su.se.bakover.service.journalføring.JournalføringOgDistribueringsResultat
+import no.nav.su.se.bakover.service.journalføring.JournalføringOgDistribueringsResultat.Companion.logResultat
+import org.slf4j.Logger
+import org.slf4j.LoggerFactory
+
+/**
+ * Distribuerer 'vanlige' dokumenter (f.eks vedtak). Ment å bli kallt fra en jobb
+ */
+class DistribuerDokumentService(
```

Comment:
Denne typen dokumenter er vel utelukkende brev? Kan man distribuere andre ting enn brev? Har vi planer om å distribuere andre ting enn brev?

Reply:
Det er bare brev som vi distribuerer yep. Sikkert ikke noen planer om å sende andre ting. Jeg liker ikke egentlig å bruke brev sånn som systemet vårt er lagt litt opp. Vi har dokumentRepo, som på mange måter også er brevRepo? Strengt tatt så sender vi ikke noen brev. Vi bestiller distribusjon på en gitt journalpost 🤷

Reply:
Dette er domenet vårt. Og da prøver vi å holde oss mest mulig unna det tekniske. Hvordan ville en veileder/saksbehandler/jurist/bruker sett på det/beskrevet denne funksjonaliteten? Har de fått distribuert en journalpost, eller har de blitt sendt et brev?

Application/infrastructure er ansvarlig for å sette opp/mappe dette til den tekniske implementasjonen. Infrastructure er også ansvarlig for den tekniske/spesifikke implementasjonen av dette.

Poenget mitt er at dette skillet bør være tydelig.